test/NuGet.Core.FuncTests/NuGet.Signing.CrossFramework.Test/CrossFrameworkVerificationTest.cs

```diff
@@ -0,0 +1,408 @@
+            {
+                var signedPackagePath = await SignedArchiveTestUtility.AuthorSignPackageAsync(
+                   authorCertificate,
+                   nupkg,
+                   dir);
+
+                var countersignedPackagePath = await SignedArchiveTestUtility.RepositorySignPackageAsync(
+                    repositoryCertificate,
+                    signedPackagePath,
+                    dir,
+                    new Uri("https://v3serviceIndex.test/api/index.json"),
+                    timestampService.Url);
+
+                // Act
+                var result = RunVerifyCommand(countersignedPackagePath);
+
+                // Assert
+                result.Success.Should().BeTrue(because: result.AllOutput);
+                result.AllOutput.Should().Contain(_successfullyVerified);
+                Regex.Matches(result.AllOutput, _noTimestamperWarning).Count.Should().Be(1);
+            }
+        }
+
+        [Fact]
+        public async Task Verify_AuthorSignedTimestampedRepositoryCounterSignedTimestampedPackage_SuccessAsync()
+        {
+            // Arrange
+            var nupkg = new SimpleTestPackageContext();
+            var defaultAuthorCertificate = await _testFixture.GetDefaultAuthorSigningCertificateAsync();
+            var defaultRepositoryCertificate = await _testFixture.GetDefaultRepositorySigningCertificateAsync();
+            var timestampService = await _testFixture.GetDefaultTrustedTimestampServiceAsync();
+
+            using (var dir = TestDirectory.Create())
+            using (var authorCertificate = new X509Certificate2(defaultAuthorCertificate))
+            using (var repositoryCertificate = new X509Certificate2(defaultRepositoryCertificate))
+            {
+                var signedPackagePath = await SignedArchiveTestUtility.AuthorSignPackageAsync(
+                   authorCertificate,
+                   nupkg,
+                   dir,
+                   timestampService.Url);
+
+                var countersignedPackagePath = await SignedArchiveTestUtility.RepositorySignPackageAsync(
+                    repositoryCertificate,
+                    signedPackagePath,
+                    dir,
+                    new Uri("https://v3serviceIndex.test/api/index.json"),
+                    timestampService.Url);
+
+                // Act
+                var result = RunVerifyCommand(countersignedPackagePath);
+
+                // Assert
+                result.Success.Should().BeTrue(because: result.AllOutput);
+                result.AllOutput.Should().Contain(_successfullyVerified);
+                Regex.Matches(result.AllOutput, _noTimestamperWarning).Count.Should().Be(0);
+            }
+        }
+
+#if IS_DESKTOP
+        //The following tests are from NuGet.Core.FuncTests\NuGet.Packaging.FuncTest\SigningTests\SignatureUtilityTests.cs.
+        //As timestamping in net5.0 is stricter, they could not be enabled in net5.0 code path. That's why we cross verify them.
```

Comment:
As discussed directly, I don't understand the purpose of the tests below. Would we consider it a bug if `dotnet nuget verify` started saying these packages are valid, just like `nuget.exe verify` presumably does?

Reply:
Thanks! I created a tracking issue https://github.com/NuGet/Client.Engineering/issues/1091, and add the link in cross framework tests. We will remove those 3 tests when we fill the gap.